### PR TITLE
Added heartbeats to wiring framework.

### LINKED ISSUE #9618: Wiring framework: add ability to schedule reocurring tasks
A small number of subsystems need the ability to periodically schedule work (e.g. the event creator and platform status manager). This functionality should be added to the wiring framework.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #9618

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]
I'm approving because the code is fairly benign.  I'm not clear yet on the over all design that the heartbeat scheduler is working towards.  Is there a diagram or plan to show this in context?

**Comment by cody-littley** [comment]
@edward-swirldslabs 
> I'm approving because the code is fairly benign. I'm not clear yet on the over all design that the heartbeat scheduler is working towards. Is there a diagram or plan to show this in context?

I've not got a diagram, but there are intended uses for this feature. Currently there are several components that need to "do something" every so often even if they don't get new input. An example of this is the event creator. If it is unable to create an event due to rate limiting, we want it to try again at a later time. This heartbeat feature will allow us to keep using this same pattern with the new wiring framework.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/HeartbeatScheduler.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    private final List<HeartbeatTask> tasks = new ArrayList<>();
+    private boolean started;
+
+    /**
+     * Constructor.
+     *
+     * @param model the wiring model containing this heartbeat scheduler
+     * @param name  the name of the heartbeat scheduler
+     */
+    public HeartbeatScheduler(@NonNull final WiringModel model, @NonNull final Time time, @NonNull final String name) {
```
Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/HeartbeatScheduler.java:50]: missing param doc for `time`

**Review by alittley — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/TaskScheduler.java [hunk 2]
```diff
@@ -65,6 +67,50 @@ protected TaskScheduler(
...
+     * @param <I>  the type of data that is inserted via this input wire
+     * @return the input wire
+     */
+    @NonNull
+    public final <I> InputWire<I, OUT> buildInputWire(@NonNull final String name) {
+        return new InputWire<>(this, name);
+    }
+
+    /**
+     * Get the default output wire for this task scheduler. Sometimes referred to as the "primary" output wire.All data
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/TaskScheduler.java:84]: fixed

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/HeartbeatScheduler.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/HeartbeatScheduler.java:50]: added

**Review by alittley — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**